tests/nrf5340_audio/lc3_file/src/main.c

```diff
@@ -0,0 +1,348 @@
+/*
+ * Copyright (c) 2024 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: LicenseRef-Nordic-5-Clause
+ */
+
+#include <zephyr/ztest.h>
+#include <zephyr/fff.h>
+#include <modules/lc3_file.h>
+#include <zephyr/fs/fs.h>
+
+DEFINE_FFF_GLOBALS;
+
+FAKE_VALUE_FUNC(int, sd_card_init);
+FAKE_VALUE_FUNC(int, sd_card_open, const char *, struct fs_file_t *);
+FAKE_VALUE_FUNC(int, sd_card_close, struct fs_file_t *);
+FAKE_VALUE_FUNC(int, sd_card_read, char *, size_t *, struct fs_file_t *);
+
+#define FRAME_BUFFER_SIZE 40
+
+static int read_lc3_fake_bytes_read;
+
+static void test_setup(void *f)
+{
+	RESET_FAKE(sd_card_init);
+	RESET_FAKE(sd_card_open);
+	RESET_FAKE(sd_card_close);
+	RESET_FAKE(sd_card_read);
+
+	FFF_RESET_HISTORY();
+
+	read_lc3_fake_bytes_read = 0;
+}
+
+int sd_card_read_header_fake_invalid(char *buf, size_t *size, struct fs_file_t *f_seg_read_entry)
+{
+	uint8_t lc3_headerarray[] = {
+		0x1c, 0xFF, /* Invalid file id, is normally 0xcc1c */
```

Comment:
Good idea! This will already be useful for the next module. Added a `mocks` folder under `tests/nrf5340_audio` that contains everything - please let me know if there's something more that could be done to make it more readable and reusable!

Reply:
Rather than mocks could it be fakes?